pkg/cloudlogging/puller.go

```diff
@@ -111,3 +113,49 @@ func (p *Puller) Pull(ctx context.Context, filter string, maxCount int) ([]*logg
 	}
 	return ls, nil
 }
+
+// SteamPull pulls live log entries, it won't stop until a cancel signal happens.
+func (p *Puller) StreamPull(ctx context.Context, filter string, logCh chan<- *loggingpb.LogEntry) (err, closeTailClientErr error) {
+	tailClient, err := p.client.TailLogEntries(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create tailLogEntriesClient: %w", err), nil
+	}
+
+	// we don't want to ignore errors from defer but if we use
+	// err = tailClient.CloseSend()
+	// this will override errors from pulling log entries,
+	// so we use a closeTailClientErr to track errors
+	// from tailClient.CloseSend().
+	defer func() {
+		closeTailClientErr = tailClient.CloseSend()
```

Comment:
Why do we need a separate error?

Reply:
As I explained in the comment, if there isn't separate error, pullErr in test will be nil and making the test fail.

Reply:
We can have one retErr, and join all the errors we want to return.

Reply:
Used errors.Join to combine the error. Now only return one error.

Reply:
+1 the join the error

Reply:
Joined.